src/Materials/Node/nodeMaterial.ts

```diff
@@ -807,33 +810,78 @@ export class NodeMaterial extends PushMaterial {
                 );
             }
 
-            // Animated blocks
-            if (this._sharedData.animatedInputs) {
-                const scene = this.getScene();
+            this._checkInternals(effect);
+        });
 
-                let frameId = scene.getFrameId();
+        return postProcess;
+    }
 
-                if (this._animationFrame !== frameId) {
-                    for (var input of this._sharedData.animatedInputs) {
-                        input.animate(scene);
-                    }
+    /**
+     * Create a new procedural texture based on this node material
+     * @param size defines the size of the texture (which will be squared)
+     * @param scene defines the hosting scene
+     * @returns the new procedural texture attached to this node material
+     */
+    public createProceduralTexture(size: number, scene: Scene): ProceduralTexture {

```

Comment:
Why not allowing non square textures, as `ProceduralTexture` does allow them?

It seems changing `number` to `any` would do the trick.

Reply:
yep correct. WIll do!